controllers/configmap_controller.go

```diff
@@ -156,6 +158,8 @@ func (r *ConfigMapReconciler) ensureInstancesAreUpToDate(instances []*instances.
 	if err != nil {
 		return err
 	}
+	windowsInstances := &core.ConfigMap{ObjectMeta: meta.ObjectMeta{Name: wiparser.InstanceConfigMap,
+		Namespace: r.watchNamespace}}
```

Comment:
Have you considered creating a separate method to do this?
```
func (r *ConfigMapReconciler) generateEvent(eventType, reason, message string){
  windowsInstances := &core.ConfigMap{ObjectMeta: meta.ObjectMeta{Name: wiparser.InstanceConfigMap,
		Namespace: r.watchNamespace}}
  		r.recorder.Eventf(windowsInstances, eventType, reason, message)
}
```
It might be beneficial to contain this logic as creating the ConfigMap seems a bit strange at first glance. I am fine with how it is currently as well, since its only being done in the `ensureInstancesAreUpToDate` and `deconfigureInstances` functions at the moment

Reply:
I am not creating a ConfigMap or are you talking about the in memory object? I don't see a way around it. Not sure how a separate method will work given I don't want to keep recreating the in memory object.

Reply:
Yeah I meant the in memory object.
Thats fine